proxyllm/provider/anthropic/claude.py

```diff
@@ -0,0 +1,168 @@
+    """
+
+    def __init__(
+        self,
+        prompt: str = "",
+        api_key: str | None = "",
+        auth_token: str | None = "",
+        temperature: float = 0.0,
+        model: str = "",
+        max_output_tokens: int | None = None,
+        timeout: int | None = None,
+    ) -> None:
+        self.prompt = prompt
+        self.api_key = api_key
+        self.auth_token = auth_token
+        self.temperature = temperature
+        self.model = model
+        self.max_output_tokens = max_output_tokens
+        self.timeout = timeout
+
+    def get_completion(self, prompt: str = "") -> str | None:
+        """
+        Requests a text completion from the specified Anthropic model.
+
+        Args:
+            prompt (str): The text prompt for generating completion.
+
+        Returns:
+            str | None: The model's text response, or None if an error occurs.
+
+        Raises:
+            AnthropicException: If an API or internal error occurs during request processing.
+        """
+        if self.api_key == "" and self.auth_token == "":
+            raise AnthropicException(
+                exception="API key or auth token not provided",
+                error_type="No API Key or Auth Token Provided",
+            )
+
+        from anthropic import Anthropic, AnthropicError
+
+        try:
+            client = Anthropic(api_key=self.api_key)
+            response = client.messages.create(
+                messages=[{"role": "user", "content": prompt or self.prompt}],
+                model=self.model,
+                max_tokens=self.max_output_tokens,
+                temperature=self.temperature,
+                timeout=self.timeout,
+            )
+        except AnthropicError as e:
+            raise AnthropicException(
+                exception=e.args[0], error_type=type(e).__name__
+            ) from e
+        except Exception as e:
+            raise AnthropicException(
+                exception=e.args[0], error_type="Unknown Anthropic Error"
+            ) from e
+        return response.content[0].text or None
+
+    def tokenize(self, prompt: str = "") -> TokenizeResponse:
+        """
+        Tokenizes the provided prompt using the tokenizer.
+
+        Args:
+            prompt (str, optional): The prompt to be tokenized. Defaults to an empty string.
+
+        Returns:
+            TokenizeResponse: An object containing information about the tokenization process,
+                including the number of input tokens and the maximum number of output tokens.
+
+        Note:
+        Note: that this is only accurate for older models, e.g. `claude-2.1`. For newer
+        models this can only be used as a _very_ rough estimate, instead you should rely
+        on the `usage` property in the response for exact counts.
+        """
+        from anthropic import Anthropic
+
+        client = Anthropic(api_key=self.api_key)
+
+        return TokenizeResponse(
```

Comment:
Does this tokenization cost credit/money? If it does, it might be best to use a different tokenizer that doesn't cost money for now; For example in the cohere tokenization function we use the general BPE tokenizer. Later we may switch it back over for the sake of accuracy, but a different of a couple of tokens should be ok

Reply:
Im pretty sure I need access to an account to check if it costs extra right?